src/lib/y2firewall/clients/proposal.rb

```diff
@@ -106,15 +108,54 @@ def write
       # Obtain and call the corresponding method for the clicked link.
       def call_proposal_action_for(link)
         action = link.gsub("firewall--", "")
-        @settings.public_send("#{action}!")
+        if action == "cpu_mitigations"
+          bootloader_dialog
+        else
+          @settings.public_send("#{action}!")
+        end
       end
 
       # Array with the available proposal descriptions.
       #
       # @return [Array<String>] services and ports descriptions
       def proposals
-        # Filter proposals with content and sort them
-        [firewall_proposal, sshd_proposal, ssh_port_proposal, vnc_fw_proposal].compact
+        # Filter proposals with content
+        [cpu_mitigations_proposal, firewall_proposal, sshd_proposal,
+         ssh_port_proposal, vnc_fw_proposal].compact
+      end
+
+      # Returns the cpu mitigation part of the bootloader proposal description
+      # Returns nil if this part should be skipped
+      # @return [String] proposal html text
+      def cpu_mitigations_proposal
+        require "bootloader/bootloader_factory"
+        bl = ::Bootloader::BootloaderFactory.current
+        return nil if bl.name == "none"
+
+        mitigations = bl.cpu_mitigations
+
+        res = _("CPU Mitigations: ") + format("<a href=\"%s\">", LINK_CPU_MITIGATIONS) +
```

Comment:
NP: I'd prefer having also the closing tag in the `format` call:

```ruby
res = _("CPU Mitigations: ") + format("<a href=\"%s\">%s</a>", LINK_CPU_MITIGATIONS, 
  mitigations.to_human_string)
```

and we should probably `html_escape` the values, just to be safe (maybe escaping just the `mitigations.to_human_string` value would be enough):

```ruby
require "erb"

res = _("CPU Mitigations: ") + format("<a href=\"%s\">%s</a>",
  ERB::Util.html_escape(LINK_CPU_MITIGATIONS), 
  ERB::Util.html_escape(mitigations.to_human_string))
```

Reply:
In fact here is format useless and can be directly included. I intentionally have it outside of translation. html escape won't work properly for href argument, for human string it makes some sense. I will adapt.